transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -153,7 +150,10 @@ public ChannelHandlerContext fireUserEventTriggered(Object event) {
     @Override
     public ChannelHandlerContext fireChannelRead(Object msg) {
         AbstractChannelHandlerContext next = findContextInbound();
-        ReferenceCountUtil.touch(msg, next);
+        if (ResourceLeakDetector.ENABLED) {
```

Comment:
@Scottmitch is epoll wait considered to use cpu or it's more like thread.sleep?
well I could just consider touching anything other than read/write syscalls a waste of time by looking at profiler 

[a](https://sc-cdn.scaleengine.net/i/a411ab99d85b37d5ff21c2b1f28e566c1.png)

but getting .touch() out is free so that's what I did